prov/acmp/src/acmp.c

```diff
@@ -1774,6 +1845,7 @@ acmp_resolve_dest(struct acmp_ep *ep, struct acm_msg *msg, uint64_t id)
 		acm_log(2, "have address, resolving route\n");
 		acm_increment_counter(ACM_CNTR_ADDR_CACHE);
 		atomic_inc(&ep->counters[ACM_CNTR_ADDR_CACHE]);
+		acmp_acquire_sgid(saddr, dest);
```

Comment:
This is how it's done in the kernel today.

If you look at drivers/infiniband/core/addr.c:resolve_addr
We call rdma_copy_addr to store the dev_addr in struct rdma_dev_addr
which will later be used in cma_acquire_dev to match the device and be in included in the GRH (if a GRH is required).

Also the way the src ip is selected today, is to create and connect a UDP socket and take the source IP from the device the socket was bound to, so today already we have a connection between the ipoib interface and the PR query
I could've add the code in acm (acm_svr_select_src) but felt we should leave it up for the provider to select the SGID.